Readme.md

```diff
@@ -1,5 +1,137 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("../src/tests/bnf/fender.bnf");
+```
+
+Then you need to setup a matcher to be able to parse the bnf
+```rust
+    let mut lexer = match bnf::parse(bnf_input){
+        Ok(v) => v,
+        Err(e) => { 
+            //Error Messages
+            return;
+        },
+    };
+```
+
+## How to set Lexer Rules
+Now that you have `lexer` we can use that to now set some rules
+```rust
+ lexer.add_rule_for_names(vec!["pow", "add", "mul", "range", "cmp", "and", "or"], CullStrategy::LiftNChildren(1));
+```
+
+# Tokens
+To tokenize your input we need to first define the result like so
+```rust
+let res = lexer.tokenize(test_input);
```

Comment:
It's not necessary to define a variable to store the result. You can inline this to something like `lexer.tokenize(test_input).unwrap()`. A readme doesn't need to include any error handling, which is what `.unwrap()` signifies - calling unwrap on an erroneous Result will panic at runtime, so it's bad practice to include in production code. But it's perfect for examples, because it communicates "this can error, but we're not handling it here since this is just an example". Rust developers will understand that's what it means when you put `.unwrap()` instead of big error handling constructs in examples.

However, it's probably still useful to mention that `FluxError` has a lot of options for debug printing to make it much nicer! `:?` for standard debug, `:#?` for pretty debug, `:#` for user-friendly, `:+#` for user-friendly with more details, if I remember correctly.